sunpy/physics/limb_darkening.py

```diff
@@ -0,0 +1,183 @@
+        New York: AIP Press; Springer, 2000. ISBN: 038798746
+    """
+
+    au = -8.9829751
+    bu = 0.0069093916
+    cu = -1.8144591e-6
+    du = 2.2540875e-10
+    eu = -1.3389747e-14
+    fu = 3.0453572e-19
+
+    a = np.array((au, bu, cu, du, eu, fu))
+
+    wavelength = wavelength.to_value()
+    wavelength_series = np.array([wavelength ** n for n in range(len(a))])
+
+    ul = np.sum(a * wavelength_series)
+
+    return ul
+
+
+def darklimb_v(wavelength):
+    """
+    Returns the constant, v, for a given wavelength for input into the limb
+    darkening function.
+
+    Parameters
+    ----------
+    wavelength : `~sunpy.map.GenericMap.wavelength`
+        The wavelength for which the limb darkening is to be calculated.
+
+    Notes
+    -----
+    Coefficients for 5th order polynomial fit to limb darkening
+    constant `v` was done on data from Astrophysical
+    Quantities by Allen.
+
+    Limitations
+    -----------
+    Only applies to wavelengths in the range 4000<lambda<15000 Angstrom
+
+    Reference
+    ---------
+    Cox, Arthur N., Allen's astrophysical quantities, 4th ed. Publisher:
+        New York: AIP Press; Springer, 2000. ISBN: 038798746
+
+    """
+
+    av = 9.2891180
+    bv = -0.0062212632
+    cv = 1.5788029e-6
+    dv = -1.9359644e-10
+    ev = 1.1444469e-14
+    fv = -2.599494e-19
+
+    a = np.array((av, bv, cv, dv, ev, fv))
+
+    wavelength = wavelength.to_value()
+    wavelength_series = np.array([wavelength ** n for n in range(len(a))])
+
+    vl = np.sum(a * wavelength_series)
+
+    return vl
+
+
+def get_limb_dark(sun_map):
+    """
+    Calculates limb darkening function for a given sun_map
+
+    Parameters
+    ----------
+    sun_map : `~sunpy.map.GenericMap`
+        The sun map for which the limb darkening is to be calculated.
+
+    Limitations
+    -----------
+    Only applies to wavelengths in the range 4000<lambda<15000 Angstrom
+
+    Returns
+    -------
+    limbfilt: `~numpy.array`
+        Numpy array representation of limb darkening function.
+
+    Notes
+    -----
+    IDL code equivalent:
+    http://hesperia.gsfc.nasa.gov/ssw/packages/nrl/idl/nrlgen/analysis/limb_dark.pro
+
+    Reference
+    ---------
+    Cox, Arthur N., Allen's astrophysical quantities, 4th ed. Publisher:
+        New York: AIP Press; Springer, 2000. ISBN: 038798746
+
+    """
+
+    if not (4000 * u.AA < sun_map.wavelength < 15000 * u.AA):
+        raise ValueError("The wavelength of the observation \
+                          must be between 4000 and 15000 Angstrom.")
+
+    ul = darklimb_u(sun_map.wavelength)
+    vl = darklimb_v(sun_map.wavelength)
+
+    x_center, y_center = sun_map.world_to_pixel(SkyCoord(0, 0, unit=u.arcsec, frame=sun_map.coordinate_frame))
+    x_center = x_center.to(u.pixel)
+    y_center = y_center.to(u.pixel)
+    x_dim = sun_map.dimensions.x.to(u.pixel).value
+    y_dim = sun_map.dimensions.y.to(u.pixel).value
+    radius = (sun_map.rsun_obs.to(u.arcsec).value /
```

Comment:
This whole section seems way too happy with de-uniting things. This radius calculation in particular doesn't need to do a `to_value` until right at the end.

I would be more interested in knowing if this whole algorithm could be made to be unitful all the way through, i.e. to `to_value` anywhere.